### DISCUSSION TITLE
Organization invites fail based on keywords such as 'lock' in the email.

### DISCUSSION THREAD

**Discussion by gigaion** [dani-garcia/vaultwarden#4432] [Q&A] [answered]
### Subject of the issue
Organization invites fail based on keywords such as 'lock' in the email.

### Deployment environment

* vaultwarden version:
  - Vaultwarden version: 1.30.5

* Install method:
  - Tested on both latest docker running on either Ubuntu or Debian. Also tested on two separate unrelated installations of vaultwarden.

* Clients used: 
  - Web vault (Join Organization Now) link

* Reverse proxy and version
  - Tested on both Nginx Proxy Manager (NPM) and Cloudflare Tunnels to verify it wasn't the reverse proxy causing the issue.

* MySQL/MariaDB or PostgreSQL version: 
  - PostgreSQL: Docker Image postgres:14.9

* Other relevant details:
  - I've discovered a temporary work around to the issue. To get around the problem and have the user join do the following:

1. Invite user to Organization like usual
2. Have them create account and use link like normal.
3. After they login and fail to join organization, have them stay on the logged in tab.
4. While still logged in, copy the invite link and paste it into the Google Chrome browser URL bar and hit enter.
5. The page will reload but remain logged in, and the link will work successfully on joining the organization.

I am unsure if this is a bug related specifically to Vaultwarden or the Web Vault code by Bitwarden. Any help by people with resources to test this on Bitwarden official clients would be beneficial in this troubleshooting.

### Steps to reproduce
- Install Vaultwarden like usual
- Create an organization
- Invite a user that contains a keyword such as 'lock' in it. In my example i did a username like testlock@input-your-domain-here.com
- When they go to register using the link, they will be unable to join the organization even after doing the registration.
- Out of 100+ people invited, there is a possibility this affects more then just the keyword 'lock', and I am actively going to start troubleshooting with another user to see if another blocked keyword exists.

### Expected behaviour
- User should be able to join organization like any other user.

### Actual behaviour
- User organization invite link does nothing for joining organization, unless you do the workaround method.

**Comment by stefan0xC** [comment]
I can't reproduce this (using Firefox). If I invite a user with a "lock" in the email address, open a new tab and open the join link, create an account and login using this join link, the organization invitation is accepted immediately.

To reproduce this I have to create an account and login without using the join link and then open the join link in the same tab. Then after selecting login and signing in again, nothing happens. Only when I then paste the join link again (in the same tab!) the invitation is accepted.

As far as I've looked into it this is an issue with the web vault that [according to this comment requires users to be in a logged out state to accept the organization](https://github.com/bitwarden/clients/blob/web-v2024.2.2/apps/web/src/app/auth/accept-organization.component.ts#L61). (And this happens regardless of a specific keyword in the user's email.)

**Comment by gigaion** [comment]
@stefan0xC that is interesting information. In my my case I've used Google Chrome to replicate the issue instead of Firefox. Tested on both windows 11 and Linux Fedora with chrome. Although in my case I used the registration link to make the account. But it was impossible to accept the organization link unless user was logged in and organization link pasted into the same tab. 

Any chance the steps are easier to replicate with chrome when you attempt it?

**Reply by BlackDex** [reply]
Even though, if we are able to replicate, it might not be an issue we can fix though. If it is client side only issue. This project does not maintain the clients.

**Reply by gigaion** [reply]
@BlackDex that is what I aim to get determined, If it is a bug with the client or vaultwarden.

If someone has a bitwarden subscription to test this on the main service, and can replicate it. this discussion will be opened over at their respective bug reporting methods.

**Reply by BlackDex** [reply]
How to test this on there main service? How to put authelia in front of it. As that you mentioned that when disabled it works.

Any useful info should be in the developer console.

**Reply by gigaion** [reply]
@BlackDex you mention authelia. By chance was that comment placed in the wrong discussion? I don't recall mentioning it working or testing with that

**Reply by BlackDex** [reply]
Indeed sorry 😐

**Reply by gigaion** [reply]
@BlackDex
I have created a Free organization on the official Bitwarden website vault, and sent an invite to testlock email. I was able to verify this does not affect the official Bitwarden website/client, as the invitation prompt worked by doing the following (Using Google Chrome):

1. Sent invitation to testlock@my-example.com
2. Clicked on 'Join Organization Now' email and created account.
3. Without refreshing, continued to login after creating account.
4. Invitation accepted notification prompt was shown, indicating successful invitation follow-through.

Although one thing I will note, is Bitwarden is using that new website UI client design, so there is a chance maybe this specific bug was fixed in the latest Bitwarden client.

**Reply by BlackDex** [reply]
I just tested it on the same version via Vaultwarden, and it works directly for me. Same as you described here.

My guess would be a bug, already fixed in the client, as nothing seems to have changed regarding the processing of this.

**Comment by BlackDex** [comment, marked as answer]
btw, i just saw a commit which might have something to do with this:
https://github.com/bitwarden/clients/commit/88f13d3be5d3c1ea7e938a64d807571df8004d86

That is fixed in the new versions, but not yet build by us. We (I) need to validate the web-vault a bit better before i'm fine with releasing it.

**Reply by stefan0xC** [reply]
Ah, no wonder I couldn't reproduce it with https://github.com/dani-garcia/bw_web_builds/pull/157 :sweat_smile:

**Reply by gigaion** [reply]
Awesome @BlackDex , thanks for tracking down that git commit!
Now that we've confirmed this is a Bitwarden bug, which will eventually streamline it's way down to Vaultwarden I'll consider this problem resolved.

In the meantime if anyone else runs into this problem, they can refer to this github discussion for the workaround method until then.
